### ISSUE TITLE
Error occurred when running `electron .`

### ISSUE THREAD

**Issue by h2appy** [platformio/platformio-atom-ide-terminal#112]
Environment:

  OSX 10.11.6
  Atom 1.12.6
  Platformio Ide Terminal 2.2.2
  Electron 1.4.10

In system terminal, it's OK when running "electron .". But in "Platformio Ide Terminal", error was returned:

module.js:457
    throw err;
    ^
Error: Cannot find module 'electron'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/un/Documents/electron-quick-start/main.js:1:97)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)

**Comment by oldham123** [comment]
+1

Works in CMD, but not here

**Comment by wuttimatic** [comment]
same error, please try `sudo electron .`

**Comment by Filipsi** [comment]
+1

Same error on Windows 10 64x, Atom 1.12.7, Electron 1.3.13, PlatformIO IDE Terminal 2.2.2.
Works in cmd and powershell, not in PlatformIO IDE Terminal.

**Comment by boredwookie** [comment]
+1

Similar behavior except that I get this error:

App threw an error during load
Error: Cannot find module 'app'
    at Module._resolveFilename (module.js:455:15)

When I run electron from a system terminal everything works fine (I'm on Lubuntu 16.04).

**Comment by DamnedScholar** [comment]
There appears to be a general error with opening any Electron applications from within Atom. I've opened an issue [on the Atom repo](https://github.com/atom/atom/issues/13624).

**Comment by Hum4n01d** [comment]
Same issue

**Comment by matthewjumpsoffbuildings** [comment]
Same issue here

**Comment by pbarbiero** [comment]
Well it might be an issue with pty.js -- I cant get it to accept the updated "env" object that "C:\Windows\system32\cmd.exe " in process.coffee creates. Its excluding env parameters from the list, but those parameters still ultimately show up in the terminal instance. Even if i manually override "ELECTRON_RUN_AS_NODE = 0", it still shows up as 1 once I run "set" in my terminal.

I think I am at my limit of tracing this. I still dont know if the issue is this plugin, atom.io, or pty.js now.

**Comment by PhillMcKraken** [comment]
I think I found it, its due to pty.js being outdated.

This project picks up where it left off, since pty.js seems to be no longer actively maintained: https://github.com/Tyriar/node-pty/releases/tag/0.4.1

Specifically, this issue: https://github.com/chjj/pty.js/issues/137

**Comment by pbarbiero** [comment]
Looks like its due to pty.js being outdated and not maintained.

https://github.com/Tyriar/node-pty has many, many fixes for windows (and other platforms as well). Investigating the code base differences between that and pty.js, there definitely are some environment fixes in there. And that is the root issue we are having here.

I am unable to test however, as when I build node-pty and switch platformio-atom-ide-terminal to use it instead, I get complaints about it not being a valid win32 application and I cant find a prebuilt version of it.

Definitely at my limits here for this sort of thing it seems like.

**Comment by pbarbiero** [comment]
Well I tried again, basing off of https://github.com/platformio/platformio-atom-ide-terminal/issues/155#issuecomment-277480500 using node-pty instead, however I get the same invalid win32 app, for both 32 and 64bit versions of Atom.

I seriously hope someone else can try and confirm if node-pty fixes these issues, as I would love to be able to run "electron ." from the embedded terminal without needing a separate instance.

A workaround is to manually do `set ELECTRON_RUN_AS_NODE=` after the terminal starts up, but I would prefer not having to do this.

**Comment by pbarbiero** [comment]
@ivankravets any input here? I unfortunately am not equipped to take it any further at this point.

**Comment by pbarbiero** [comment]
@ivankravets 
As I said in https://github.com/platformio/platformio-atom-ide-terminal/issues/112#issuecomment-282343098, the issue on windows is pty.js is outdated, and the "env" variable updated in that pull does *NOT* get passed to the terminal instance -- that is ignored.

That was one of the very first things I tried unfortunately. This will need to be reopened, along with many of those other issues I am sure -- atleast for windows.

**Comment by pbarbiero** [comment]
@ivankravets  Confirmed still broken in 2.4.0

**Comment by ivankravets** [comment]
@PhillMcKraken Thanks, will wait for a PR to fix this issue.

**Comment by pbarbiero** [comment]
@ivankravets I finally managed to figure out why my build was failing, I wasnt running apm rebuild properly to build the native modules properly for the current atom/electron revision.

I did confirm that this issue was resolved entirely by switching to node-pty (not to mention it is much, MUCH faster as well, along with a boatload of fixes for windows in general)

I can make a pull, but all I did was change pty.js to node-pty and included it instead of pty.js from process.coffee (see https://github.com/PhillMcKraken/platformio-atom-ide-terminal/commit/5653f119d225f09cd6053d5b9ea1b267542a228c) -- the issue here is that requires everyone to have python2.8 installed on their machine (everything else seems to be already included with the apm processes) I noticed you did some prebuild work with pty.js, that would have to be done for this as well.

I was unable to make an appveyor build work, even following your process in https://github.com/platformio/platformio-atom-ide-terminal/issues/155#issuecomment-277480500
I can make a pull request now with that minor change as detailed above, but that wont help most people and probably would break it for more people without shipping it as a prebuilt version.

Any suggestions?
